src/main/steps/applicant2/address-private/post.ts

```diff
@@ -1,3 +1,20 @@
+import autobind from 'autobind-decorator';
+
+import { Case, CaseWithId } from '../../../app/case/case';
+import { YesOrNo } from '../../../app/case/definition';
+import { AppRequest } from '../../../app/controller/AppRequest';
+import { AnyObject } from '../../../app/controller/PostController';
 import CitizenUpdateContactDetailsPostController from '../../applicant1/check-phone-number/post';
 
-export default CitizenUpdateContactDetailsPostController;
+@autobind
+export default class CitizenUpdateContactDetailsPostControllerApp2WithRefuge extends CitizenUpdateContactDetailsPostController {
+  protected async save(req: AppRequest<AnyObject>, formData: Partial<Case>, eventName: string): Promise<CaseWithId> {
+    const setInRefugeDefault =
+      req.session.userCase.applicant2AddressPrivate === YesOrNo.NO ||
+      req.session.userCase.applicant2InRefuge === undefined
+        ? { applicant2InRefuge: YesOrNo.NO }
+        : {};
```

Comment:
I thought we might be able to do this without the controller code using `to-api-format.ts`, e.g. by setting a value for `applicant2InRefuge` when `applicant2AddressPrivate` is altered:
https://github.com/hmcts/nfdiv-frontend/pull/4004/files#diff-bacbc2e038fda9d76b87b53b2ef1c7abba35a7841242ff8b7c1aff62b1a4a0a8R147

Reply:
It looked like we have specific post controllers for other pages, that's why I plumped for putting it there